borgmatic/config/schema.yaml

```diff
@@ -88,6 +88,13 @@ map:
                 type: scalar
                 desc: Umask to be used for borg create.
                 example: 0077
+            archive_name:
```

Comment:
I might suggest calling this `archive_name_format` to make clear that it's the pattern for archive names, rather than the literal archive name itself. (This naming idea is from an earlier patch from @squires.)